### PR TITLE
icon-loader: add a log handler for sfdo messages

### PR DESCRIPTION
**Pull request by jlindgren90**
Useful for debugging why an icon isn't being loaded.

### CODE REVIEW

File: src/icon-loader.c [hunk 1]
```diff
@@ -25,6 +25,18 @@ struct icon_loader {
 	struct sfdo_icon_theme *icon_theme;
 };
 
+static void
+log_handler(enum sfdo_log_level level, const char *fmt, va_list args, void *data)
+{
+	// add a prefix if the format length is reasonable
+	char buf[256];
+	if (snprintf(buf, sizeof(buf), "[sfdo] %s", fmt) < (int)sizeof(buf)) {
```
Comment (tokyo4j) [inline src/icon-loader.c:33]: I suggest setting prefixes like `[sfdo-desktop]` and `[sfdo-icon]`. I think the argument `data` can be used for this.

File: src/icon-loader.c [hunk 2]
```diff
@@ -25,6 +25,18 @@ struct icon_loader {
 	struct sfdo_icon_theme *icon_theme;
 };
 
+static void
+log_handler(enum sfdo_log_level level, const char *fmt, va_list args, void *data)
+{
+	// add a prefix if the format length is reasonable
```
Comment (tokyo4j) [inline src/icon-loader.c:31]: nit: `/* */` should be used for comments.

**Comment by tokyo4j** [comment]
LGTM other than my minor comments.

**Comment by johanmalm** [comment]
Nice. Thanks.

**Pull request merged**